### ISSUE TITLE
Getopt::Long minimum version 2.39 please 

### ISSUE THREAD

**Issue by n1vux** [beyondgrep/ack3#193]
Running Ack 3 beta 6 under a vanilla Perl 5.10.1 (with only Test::Class added in perlbrew cpanm) , it fails as follows

*`Can't locate object method "getoptionsfromarray" via package "Getopt::Long::Parser" at /home/wdr/bin/ack line 2755.`* 

Since the --help is done with a different Getopt api, this one is only used in App::Ack::ConfigLoader, it managed to do `ack --version` without hitting that
```
ack 2.999_06 (standalone version)
Running under Perl 5.10.1 at /home/wdr/perl5/perlbrew/perls/perl-5.10.1/bin/perl

Copyright 2005-2019 Andy Lester.

This program is free software.  You may modify or distribute it
under the terms of the Artistic License v2.0.
```
*`corelist`* says 5.10.1 had `Getopt::Long 2.38`

Getopt::Long CHANGES says `GetOptionsFromArray`  in camelcase was added in 2.36, exported on demand only, on the main API; the lower case version is in the parent class as a local then but that was not  that is notwhen it was added to the OO interface as a direct callable lower case call.

EIther we change the call to use the Getopt::Long::GetOptionsFromArray (importable on request) or change the minimum version to 2.39 when this lowercase function first appears in Getopt::Long::Parser.

**Comment by n1vux** [comment]
I'm inferring that the OO interface Getopt::Long::Parser was rushed in to 5.10, and it wasn't quite stable  

I confirm that 2.39 is adequate not only to find the needed method but actually load config, as I did
```
perlbrew use perl-5.10.1@Test
cpanm https://cpan.metacpan.org/authors/id/J/JV/JV/Getopt-Long-2.39.tar.gz
ack 'getoptionsfromarray|Getopt::Long::Parser'
```
successfully. So my Ack3 is fixed in my 5.10 Test::Class sandbox.

**Comment by petdance** [comment]
Done in 87107ae.  Thanks for the research.
